app/Http/Controllers/TaskController.php

```diff
@@ -2,34 +2,47 @@
 
 namespace App\Http\Controllers;
 
+use Auth;
+use Session;
+use App\Models\Task;
 use Illuminate\Http\Request;
-
+use App\Models\TaskCategory;
+use App\Http\Requests\TaskStoreRequest;
 use Illuminate\Support\Facades\Validator;
 
-use App\Models\Task;
-
 class TaskController extends Controller
 {
     /**
      * Display a listing of the resource.
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $taskList = Task::all();
-        return view('Task.dashboard',compact('taskList'));
+        if (isset($request->filter_id)) {
+            $taskList = Task::where(['user_id' => Auth::user()->id, 'category_id' => $request->filter_id])->get();
```

Comment:
Use relationship to find current user's task.

Reply:
ok sir.